pkg/openapiclient/local_crds.go

```diff
@@ -33,55 +33,60 @@ var metadataSchemas map[string]*spec.Schema = func() map[string]*spec.Schema {
 
 // client which provides openapi read from files on disk
 type localCRDsClient struct {
-	fs  fs.FS
-	dir string
+	fs   fs.FS
+	dirs []string
 }
 
 // Dir should have openapi files following directory layout:
 // myCRD.yaml (groupversions read from file)
-func NewLocalCRDFiles(fs fs.FS, dirPath string) openapi.Client {
+func NewLocalCRDFiles(fs fs.FS, dirPaths []string) openapi.Client {
 	return &localCRDsClient{
-		fs:  fs,
-		dir: dirPath,
+		fs:   fs,
+		dirs: dirPaths,
 	}
 }
 
 func (k *localCRDsClient) Paths() (map[string]openapi.GroupVersion, error) {
-	if len(k.dir) == 0 {
+	if len(k.dirs) == 0 {
 		return nil, nil
 	}
-	files, err := utils.ReadDir(k.fs, k.dir)
-	if err != nil {
-		return nil, fmt.Errorf("error listing %s: %w", k.dir, err)
-	}
 	var documents []utils.Document
-	for _, f := range files {
-		path := filepath.Join(k.dir, f.Name())
-		if f.IsDir() {
-			continue
+
+	for _, dir := range k.dirs {
+		files, err := utils.ReadDir(k.fs, dir)
+		if err != nil {
+			return nil, fmt.Errorf("error listing %s: %w", dir, err)
```

Comment:
Do we want to stop if any of the directory fails? I think we do
But do we want to report all errors if multiple directories fail? I suspect so too, thoughts?

Reply:
I agree for a better user experience we likely should accumulate and report all errors at once.
However it seems the current strategy for this function is to fail as soon as any error is encountered (e.g. first file that cannot be read). Perhaps we could revisit it outside of this PR ?

Reply:
sgtm